code/modules/codex/categories/category_reagents.dm

```diff
@@ -10,9 +10,34 @@
 			continue
 		var/chem_name = lowertext(initial(reagent.name))
 		var/datum/codex_entry/entry = new( \
-		 _display_name = "[chem_name] (chemical)", \
-		 _associated_strings = list("[chem_name] pill"), \
-		 _lore_text = "[initial(reagent.description)] It apparently tastes of [initial(reagent.taste_description)].")
+		_display_name = "[chem_name] (chemical)", \
+		_associated_strings = list("[chem_name] pill"), \
+		_lore_text = "[initial(reagent.description)] It apparently tastes of [initial(reagent.taste_description)].", \
+		_mechanics_text = "")
+
+		switch(initial(reagent.reagent_state))
+			if(SOLID)
+				entry.mechanics_text += "It's a solid.</br>"
+			if(LIQUID)
+				entry.mechanics_text += "It's a liquid.</br>"
+			if(GAS)
+				entry.mechanics_text += "It's a gas.</br>"
+
+		if(initial(reagent.color_foods))
+			entry.mechanics_text += "It can be used to dye foods.</br>"
+
+		if(initial(reagent.overdose))
+			entry.mechanics_text += "This causes adverse effects if more than [initial(reagent.overdose)]u is ingested."
+
+		if(initial(reagent.metabolism) != REM)
+			entry.mechanics_text += "It metabolizes [(initial(reagent.metabolism) < REM) ? "slower" : "quicker"] than the average reagent.<br>"
+
+		// since chilling/heating points are represented in kelvin, we need to subtract to get in celsius
+		if(initial(reagent.chilling_point))
+			entry.mechanics_text += "At [(initial(reagent.chilling_point) - T0C)] degrees celsius, this chills into [initial(reagent.chilling_prod_english)].<br>"
+
+		if(initial(reagent.heating_point))
+			entry.mechanics_text += "At [(initial(reagent.heating_point) - T0C)] degrees celsius, this boils into [initial(reagent.heating_prod_english)].<br>"
```

Comment:
just get reagent's names from list instead of hardcoding results as string

Reply:
> just get reagent's names from list instead of hardcoding results as string

Yeah, I spent several hours trying to get that to work. My understanding is that, while normal integers and strings are matched to datums on compile, lists are created on actual datum creation, and thus trying to access the list either:
1) fails (if you just try and read the list, it throws an error)
2) returns nothing (if you use initial)
The only way of bypassing this that I saw was actually creating the reagent, but to do that I would have to create a container datum, which requires me to create an atom for it to reside in, and so on. Maybe I'm missing something, but I really didn't see a way around it other than hardcoding it.